backend-project/small_eod/letters/serializers.py

```diff
@@ -19,12 +37,52 @@ class Meta:
             "ordering",
             "comment",
             "excerpt",
+            "description",
         ]
 
+    def create(self, validated_data):
+        descriptions = validated_data.pop("description")
 
-class DescriptionSerializer(serializers.ModelSerializer):
-    class Meta:
-        model = Description
-        fields = [
-            "name",
+        validated_data["address"] = Description.objects.create(
+            **validated_data.pop("address")
+        )
+        validated_data["channel"] = Description.objects.create(
+            **validated_data.pop("channel")
+        )
+        validated_data["description"] = Description.objects.create(
+            **validated_data.pop("description")
+        )
+        institution = validated_data.pop("institution")
+        case = validated_data.pop("case")
+
+        letter = super().create(validated_data)
+
+        for description in descriptions:
+            Description.objects.get_or_create(name=description, letter=letter)
+
+        letter.institution = institution
+        letter.case = case
+        letter.save()
+        return letter
+
+    def update(self, instance, validated_data):
+        """
+        nested - variable storing representations of the nested objects
+        of LetterSerializer (Channel, Address and Description).
+        Iterating over those 3 and updating fields of the related objects,
+        using key-value pairs from PATCH request.
+        """
+        nested = [
+            {"instance": instance.address, "data": validated_data.pop("address", {})},
+            {"instance": instance.channel, "data": validated_data.pop("channel", {})},
+            {
+                "instance": instance.description,
```

Comment:
To tak na pewno nie przejdzie - bede zmienial metode 'update', zeby byla w stanie ogarnac duzo 'description'. Tylko  w jaki sposob chcemy zarzadzac opisami listu? PATCH /letters/<pk>?  czy moze POST/DELETE /letters/<pk>/description

Reply:
https://github.com/watchdogpolska/small_eod/blob/dev/docs/swagger.yaml#L1281-L1367

Reply:
Dlaczego chcesz dużo ```description```?  Jeden mamy obecnie i jeden wystarczy...